### PR TITLE
[data types] Add dtypes to factoring bloqs.

### PR DESCRIPTION
**Pull request by fdmalone**
Working towards #446

### CODE REVIEW

File: qualtran/bloqs/factoring/mod_add.py [hunk 1]
```diff
@@ -48,9 +48,9 @@ class CtrlScaleModAdd(Bloq):
     def signature(self) -> 'Signature':
         return Signature(
             [
-                Register('ctrl', bitsize=1),
-                Register('x', bitsize=self.bitsize),
-                Register('y', bitsize=self.bitsize),
+                Register('ctrl', bitsize=QBit()),
+                Register('x', bitsize=QAny(self.bitsize)),
+                Register('y', bitsize=QAny(self.bitsize)),
```
Comment (mpharrigan) [inline qualtran/bloqs/factoring/mod_add.py:53]: how should I be reviewing this PR? I would expect this to be `QInt` unless this PR is only part of the migration

Comment (fdmalone) [reply qualtran/bloqs/factoring/mod_add.py:53]: No this is actually a mistake. Are these unsigned or signed integers?

Comment (fdmalone) [reply qualtran/bloqs/factoring/mod_add.py:53]: Unsigned I think right?

**Comment by fdmalone** [comment]
I made a flying change to reflect #632 since it's the first use of QUInt

File: qualtran/bloqs/factoring/mod_add.py (hunk 1 above)
Comment (mpharrigan) [reply qualtran/bloqs/factoring/mod_add.py]: yes

**Review by mpharrigan — APPROVED** [review]

**Pull request merged**